homeassistant/core.py

```diff
@@ -1410,6 +1410,8 @@ def __repr__(self) -> str:
 class StateMachine:
     """Helper class that tracks the state of different entities."""
 
+    __slots__ = ("_states", "_reservations", "_bus", "_loop")
```

Comment:
What's the benefit with slots here?

Reply:
Usually it's for memory, but slotted classes also have faster attribute lookups since they don't have the overhead of `__dict__`